tests/integration/proxy/grpcproxy/cluster_test.go

```diff
@@ -67,6 +71,40 @@ func TestClusterProxyMemberList(t *testing.T) {
 	if mresp.Members[0].ClientURLs[0] != cts.caddr {
 		t.Fatalf("mresp.Members[0].ClientURLs[0] expected %q, got %q", cts.caddr, mresp.Members[0].ClientURLs[0])
 	}
+
+	//test proxy member up/down
+	newMemberAddr := "127.0.0.2:6789"
+	grpcproxy.Register(lg, cts.c, prefix, newMemberAddr, 7)
+	// wait some time for proxy update members
+	time.Sleep(time.Second)
+
+	//check add member succ
+	mresp, err = client.Cluster.MemberList(context.Background())
+	if err != nil {
+		t.Fatalf("err %v, want nil", err)
+	}
+	if len(mresp.Members) != 2 {
+		t.Fatalf("len(mresp.Members) expected 2, got %d (%+v)", len(mresp.Members), mresp.Members)
+	}
+	if mresp.Members[1].ClientURLs[0] != newMemberAddr {
```

Comment:
nit: I think `Members` is populated from a [map](https://github.com/etcd-io/etcd/blob/9a922091ed07c56c0c09575c2328e9576c76d942/server/proxy/grpcproxy/cluster.go#L142), so order isn't [guarantied](https://go.dev/blog/maps). Might cause test flakiness. 

Can we add a check for Members[0] and add a comment about order?

Reply:
map can not guarantee order, we should check if the new member is in `mresp.Member`, codes have updated, PTAL